processing/src/main/java/org/apache/carbondata/processing/merger/CarbonDataMergerUtil.java

```diff
@@ -311,10 +311,8 @@ public static String getLoadNumberFromLoadName(String loadName) {
       listOfSegmentsToBeMerged = identifySegmentsToBeMergedBasedOnSize(compactionSize,
               listOfSegmentsLoadedInSameDateInterval, carbonLoadModel);
     } else {
-
-      listOfSegmentsToBeMerged =
-              identifySegmentsToBeMergedBasedOnSegCount(listOfSegmentsLoadedInSameDateInterval,
-                      tableLevelProperties);
+      listOfSegmentsToBeMerged = identifySegmentsToBeMergedBasedOnSegCount(compactionSize,
```

Comment:
may be you can calculate compaction size, inside this method as table property is already there. No need to modify this method signature.